### ISSUE TITLE
Destroy index out of bounds 

### ISSUE THREAD

**Issue by th3byrdm4n** [DataTables/Buttons#91]
Environment: 
* DataTables 1.10.12
* Buttons 1.2.0

When I call datatables.destroy(), I would run into _undefined_ buttons[i] error caused by 

```
// Destroy the instance, cleaning up event handlers and removing DOM
destroy: function () {
  var buttons = this.s.buttons;
  [...]
  for (i = 0, ien = buttons.length; i < ien; i++) {
    this.remove(buttons[**i**].node); 
  }
```
This function is removing items in a list it's iterating over ...  Quick fix =>
`this.remove(buttons[**0**].node);`

remove the first element until the list is gone.

Somewhat interesting (I didn't dive into it); is this bug would only occur when I had my default preferences included (any info on that?):
```
$.extend(true, $.fn.dataTable.defaults, {
    dom: 'Bfrtlip',
    lengthMenu: [[10, 25, 50, -1], [10, 25, 50, "All"]],
    buttons: [
        'copy', 'excel', 'csv'
    ]
});
```

**Comment by DataTables** [comment]
Thanks for your message. This was fixed in Buttons 1.2.1 which was released yesterday. Can you try that version please

**Comment by dtopaloglou** [comment]
I finally found this bug's solution here! Thank you! :+1: @DataTables
